cmake/SpectreSetupPythonPackage.cmake

```diff
@@ -44,6 +44,12 @@ configure_file(
   "${CMAKE_BINARY_DIR}/tmp/LoadPython.sh"
   "${CMAKE_BINARY_DIR}/bin/LoadPython.sh")
 
+# Copy the Python package to CMake's install directory
```

Comment:
Would you mind expanding why this is necessary in the comment? Given how finicky installs can be with matching version numbers, etc. I think that recording what issues (or potential) issues this mitigates would be a good idea

Reply:
I added a user option to set the install dir for the Python package, and some docs. I can't really make the docs more precise than they are now I think, because it really depends on where the user wants the package installed. For example, Spack expects the package in `install_prefix/lib/pythonX.Y/site-packages`, which is the convention and the default that I chose. The same applies if the install prefix is something like `/usr` or `/usr/local` (i.e., system-wide install, such as perhaps in a container). The user may also change the `SPECTRE_PYTHON_INSTALL_LIBDIR` to the location of some virtualenv, or wherever they want the Python package to get installed.